File: cpp/ex3_1_1.cpp
#include <string>

using namespace std;

template <typename T> T choose(T v1, T v2) { return true ? v1 : v2; }

template <typename T> T mult(T v1, T v2, T v3) { return v1 * v2 * v3; }

int main() {
    string str = choose<string>("Korean", "Foreigner");
    int num = choose<int>(1, 2);
    mult<int>(1, 2, 3);
}

File: go/ex3_1_1.go
package main

func choose[T any](v1 T, v2 T) T {
    if true {
        return v1
    } else {
        return v2
    }
}

func main() {
    var str string = choose[string]("Korean", "Foreigner")
    var num int = choose[int](1, 2)
    _ = str
    _ = num
}
